<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>YubiKeyConfigForm</class>
 <widget class="QWidget" name="YubiKeyConfigForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>500</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>YubiKey OATH Configuration</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="devicesGroupBox">
     <property name="title">
      <string>YubiKey Devices</string>
     </property>
     <layout class="QVBoxLayout" name="devicesLayout">
      <item>
       <widget class="QQuickWidget" name="qmlWidget">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>200</height>
         </size>
        </property>
        <property name="resizeMode">
         <enum>QQuickWidget::SizeRootObjectToView</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="displayGroupBox">
     <property name="title">
      <string>Display Options</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="notificationsLabel">
        <property name="text">
         <string>Notifications:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QCheckBox" name="showNotificationsCheckbox">
        <property name="text">
         <string>Show system notifications</string>
        </property>
        <property name="toolTip">
         <string>Display notifications for YubiKey touch requirements and copied codes</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="usernameLabel">
        <property name="text">
         <string>Show username:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QCheckBox" name="showUsernameCheckbox">
        <property name="text">
         <string>Display username in credential names</string>
        </property>
        <property name="toolTip">
         <string>Show username in parentheses after issuer name</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="codeLabel">
        <property name="text">
         <string>Show code:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QCheckBox" name="showCodeCheckbox">
        <property name="text">
         <string>Display code if no touch required</string>
        </property>
        <property name="toolTip">
         <string>Show TOTP/HOTP code in search results for credentials that don't require touch</string>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="deviceNameLabel">
        <property name="text">
         <string>Show device name:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QCheckBox" name="showDeviceNameCheckbox">
        <property name="text">
         <string>Display YubiKey device name</string>
        </property>
        <property name="toolTip">
         <string>Show the name of the YubiKey device in credential display</string>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QWidget" name="indentedCheckboxContainer" native="true">
        <layout class="QHBoxLayout" name="indentedLayout">
         <property name="leftMargin">
          <number>30</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QCheckBox" name="showDeviceNameOnlyWhenMultipleCheckbox">
           <property name="text">
            <string>Only when multiple devices connected</string>
           </property>
           <property name="toolTip">
            <string>Only show device name when more than one YubiKey is connected</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="notificationExtraTimeLabel">
        <property name="text">
         <string>Extra notification time:</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QSpinBox" name="notificationExtraTimeSpinbox">
        <property name="toolTip">
         <string>Additional time to keep code notification visible after TOTP code expires</string>
        </property>
        <property name="suffix">
         <string> seconds</string>
        </property>
        <property name="minimum">
         <number>0</number>
        </property>
        <property name="maximum">
         <number>30</number>
        </property>
        <property name="value">
         <number>15</number>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="primaryActionLabel">
        <property name="text">
         <string>Primary action:</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QComboBox" name="primaryActionCombo">
        <property name="toolTip">
         <string>Default action when pressing Enter on a match</string>
        </property>
        <item>
         <property name="text">
          <string>Copy to clipboard</string>
         </property>
         <property name="userData" stdset="0">
          <string>copy</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Type code</string>
         </property>
         <property name="userData" stdset="0">
          <string>type</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="timeoutLabel">
        <property name="text">
         <string>Touch timeout:</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QSpinBox" name="touchTimeoutSpinbox">
        <property name="toolTip">
         <string>Maximum time to wait for YubiKey touch confirmation (0 = no timeout)</string>
        </property>
        <property name="suffix">
         <string> seconds</string>
        </property>
        <property name="specialValueText">
         <string>No timeout</string>
        </property>
        <property name="minimum">
         <number>0</number>
        </property>
        <property name="maximum">
         <number>60</number>
        </property>
        <property name="value">
         <number>10</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="portalPermissionsGroupBox">
     <property name="title">
      <string>Portal Permissions</string>
     </property>
     <layout class="QVBoxLayout" name="portalLayout">
      <item>
       <widget class="QLabel" name="portalPermissionsDescription">
        <property name="text">
         <string>Grant permanent access to XDG Desktop Portal features to avoid permission dialogs:</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="screenshotPermissionCheckbox">
        <property name="text">
         <string>Allow automatic screenshots</string>
        </property>
        <property name="toolTip">
         <string>Grant permission to capture screenshots without prompts (required for Add OATH workflow)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="remoteDesktopPermissionCheckbox">
        <property name="text">
         <string>Allow automatic text input</string>
        </property>
        <property name="toolTip">
         <string>Grant permission to emulate keyboard input without prompts (required for auto-type feature)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="portalPermissionsWarning">
        <property name="text">
         <string>&lt;small&gt;&lt;i&gt;Note: These permissions affect all applications, not just this plugin. Enable only if you trust the applications using these features.&lt;/i&gt;&lt;/small&gt;</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QQuickWidget</class>
   <extends>QWidget</extends>
   <header>QQuickWidget</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>showDeviceNameCheckbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>showDeviceNameOnlyWhenMultipleCheckbox</receiver>
   <slot>setEnabled(bool)</slot>
  </connection>
 </connections>
</ui>